fix: background image clipping on extra wide screens

aswin-s opened the pull request:
<!-- If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit. -->

### Details
<!-- Explanation of the change or anything fishy that is going on -->

Background image used in Report page and Request Money page looks clipped on extra wide screens.

### Fixed Issues
<!---
1. Please postfix `$` with a URL link to the GitHub issue this Pull Request is fixing. For example, `$ https://github.com/Expensify/App/issues/<issueID>`.
2. Please postfix  `PROPOSAL:` with a URL link to your GitHub comment, which contains the approved proposal (i.e. the proposal that was approved by Expensify).  For example, `PROPOSAL: https://github.com/Expensify/App/issues/<issueID>#issuecomment-1369752925`

Do NOT add the special GH keywords like `fixed` etc, we have our own process of managing the flow.
It MUST be an entire link to the github issue and your comment proposal ; otherwise, the linking and its automation will not work as expected.

Make sure this section looks similar to this (you can link multiple issues using the same formatting, just add a new line):

$ https://github.com/Expensify/App/issues/<issueID>
$ https://github.com/Expensify/App/issues/<issueID(comment)>

Do NOT only link the issue number like this: $ #<issueID>
--->
$ https://github.com/Expensify/App/issues/23477
PROPOSAL: https://github.com/Expensify/App/issues/23477#issuecomment-1668845896


### Tests
<!---
Add a numbered list of manual tests you performed that validates your changes work on all platforms, and that there are no regressions present.
Add any additional test steps if test steps are unique to a particular platform.
Manual test steps should be written so that your reviewer can repeat and verify one or more expected outcomes in the development environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->
1. Sign in to the web app on a desktop browser
2. Navigate to a new chat page or request money page
3. Check out the background image on the page header
4. Top and bottom portion of background image shouldn't be clipped even if user resize browser size
5. For browser width greater than 2000px the background image should repeat horizontally
6. Repeat same steps on Desktop app as well.
7. iOS & Android apps should behave exactly as before

- [x] Verify that no errors appear in the JS console

### Offline tests
<!---
Add any relevant steps that validate your changes work as expected in a variety of network states e.g. "offline", "spotty connection", "slow internet", etc. Manual test steps should be written so that your reviewer and QA testers can repeat and verify one or more expected outcomes. If you are unsure how the behavior should work ask for advice in the `#expensify-open-source` Slack channel.
--->
1. Sign in to the web app on a desktop browser
2. Navigate to a new chat page or request money page
3. Check out the background image on the page header
4. Top and bottom portion of background image shouldn't be clipped even if user resize browser size
5. For browser width greater than 2000px the background image should repeat horizontally
6. Repeat same steps on Desktop app as well.
7. iOS & Android apps should behave exactly as before

- [x] Verify that no errors appear in the JS console


### QA Steps
<!---
Add a numbered list of manual tests that can be performed by our QA engineers on the staging environment to validate that your changes work on all platforms, and that there are no regressions present.
Add any additional QA steps if test steps are unique to a particular platform.
Manual test steps should be written so that the QA engineer can repeat and verify one or more expected outcomes in the staging environment.

For example:
1. Click on the text input to bring it into focus
9. Upload an image via copy paste
10. Verify a modal appears displaying a preview of that image
--->

1. Sign in to the web app on a desktop browser
2. Navigate to a new chat page or request money page
3. Check out the background image on the page header
4. Top and bottom portion of background image shouldn't be clipped even if user resize browser size
5. For browser width greater than 2000px the background image should repeat horizontally
6. Repeat same steps on Desktop app as well.
7. iOS & Android apps should behave exactly as before

- [x] Verify that no errors appear in the JS console

### PR Author Checklist
<!--
This is a checklist for PR authors. Please make sure to complete all tasks and check them off once you do, or else your PR will not be merged!
-->

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

<!-- add screenshots or videos here -->


https://github.com/Expensify/App/assets/6880914/0a63a4fc-86e3-4ec7-94f0-f5b02e79943e



</details>

<details>
<summary>Mobile Web - Chrome</summary>

<!-- add screenshots or videos here -->

https://github.com/Expensify/App/assets/6880914/0d8899ec-71d4-40d5-88d9-7b9813fc37d4


</details>

<details>
<summary>Mobile Web - Safari</summary>

<!-- add screenshots or videos here -->


https://github.com/Expensify/App/assets/6880914/d0ba3459-48cc-47fa-97e6-fa6c40b63cfc



</details>


<details>
<summary>Desktop</summary>

<!-- add screenshots or videos here -->


https://github.com/Expensify/App/assets/6880914/edc18399-fbdc-4d3d-ae05-7885b5186cbd




</details>

<details>
<summary>iOS</summary>

<!-- add screenshots or videos here -->


https://github.com/Expensify/App/assets/6880914/c3a431f9-d1dc-4fd9-8dde-1f454b96e690



</details>


<details>
<summary>Android</summary>

<!-- add screenshots or videos here -->

https://github.com/Expensify/App/assets/6880914/5c3af58b-2424-450c-b109-d36b530674e6


</details>

aswin-s committed:
fix: background image clipping on extra wide screens

aswin-s committed:
fix: limit resizeMode repeat to web platforms

aswin-s committed:
fix: whitespace before prop name

aswin-s committed:
fix: clean lint

abdulrahuman5196 commented on src/components/ReportActionItem/MoneyReportView.js:
```diff
@@ -26,17 +28,24 @@ const propTypes = {
+    // If window width is greater than the max background width, repeat the background image
+    const maxBackgroundWidth = variables.sideBarWidth + CONST.EMPTY_STATE_BACKGROUND.ASPECT_RATIO * CONST.EMPTY_STATE_BACKGROUND.WIDE_SCREEN.IMAGE_HEIGHT;
+    const platform = getPlatform();
+    const isWebPlatform = [CONST.PLATFORM.WEB, CONST.PLATFORM.DESKTOP].includes(platform);
```
@aswin-s Why do we need platforms check? Regardless of platform the solution is specific to image size and screen size, right?

aswin-s replied:
@abdulrahuman5196  We are using Image component from 'react-native-fast-image' in these screens. FastImage doesn't support 'repeat' resize mode. So I just limited it to Web based platforms. The other option is to use Image component from 'react-native'. If you think that's ok I'll make the switch and remove the platform checks.

aswin-s replied:
Oh I just checked again. Fast image is used only on 'MoneyRequestView' page. I'll switch out the component in that screen and remove the check all together.

aswin-s committed:
fix: remove platform check for web

aswin-s commented:
@abdulrahuman5196 Removed the additional check by reverting to Image component from 'react-native' which supports [resize mode repeat](https://reactnative.dev/docs/image#resizemode).

abdulrahuman5196 commented on src/components/ReportActionItem/MoneyRequestView.js:
```diff
@@ -23,10 +23,10 @@ import useLocalize from '../../hooks/useLocalize';
 import * as ReceiptUtils from '../../libs/ReceiptUtils';
 import useWindowDimensions from '../../hooks/useWindowDimensions';
 import transactionPropTypes from '../transactionPropTypes';
-import Image from '../Image';
```
@aswin-s 

This also uses react native Image internally. You can just pass the params and it should work. Not sure why we need to change the component?

aswin-s replied:
@abdulrahuman5196 Nope. For native components we are importing Image component from 'react-native-fast-image'

https://github.com/Expensify/App/blob/9398754f3add40d84c9c84c23b79f3abb6165fa2/src/components/Image/index.native.js#L2

aswin-s replied:
If we pass resize mode as 'repeat' for Image component from FastImage, it throws error on native platforms

![Screenshot_1693328280](https://github.com/Expensify/App/assets/6880914/bcde5f19-e900-4b1a-8bf3-2e299fd92a12)

abdulrahuman5196 replied:
@aswin-s 
Oh got it. For native components we are using fast image. But still I wouldn't want us to change the components to fix this particular issue. 

We should explore on how to send proper values based on platform.(Not the previous code change to check platforms from array). Maybe we can do some tweaking in index.native.js

aswin-s replied:
But both the other places are using Image component from react-native to show the background image. So it is not a hack just to fix this issue. And since we are embedding the background image, none of the advantages of using fast-image (like caching) are useful in this case. Also note that [request for support](https://github.com/DylanVann/react-native-fast-image/issues/253) of repeat mode in fast image is open since 2018.

abdulrahuman5196 replied:
@grgia Could you kindly confirm on this? We have been waiting on this for sometime?

grgia replied:
@abdulrahuman5196 @aswin-s 

I'd agree with @aswin-s 
> The only benefit of using fast-image library is improving load performance of remote images by caching. Which is not the case here. Background image is embedded with the binary

I think we can use the native library for this case

aswin-s committed:
Merge branch 'main' into fix/issue-23477

abdulrahuman5196 commented:
@aswin-s Single transaction page's background image is still clipped. Could you kindly check on the same?


https://github.com/Expensify/App/assets/46707890/2de75e5f-1586-454b-be48-2bd8c1189f77

aswin-s committed:
fix: windowWidth is getting undefined

aswin-s committed:
fix: redundant change

aswin-s commented:
@abdulrahuman5196 Fixed. `withWindowDimensions` HOC was removed recently after refactoring the component to functional component. Now using `windowWidth` value from `useDimensions` hook.

![localhost_8082_r_4796714737810116](https://github.com/Expensify/App/assets/6880914/1b336029-835f-4bb8-8cf1-dc0b9929758a)

aswin-s committed:
fix: clean lint

abdulrahuman5196 commented:
@aswin-s , For some reason, the background image is flickering, I think first the image is taking `cover` and then changing to `repeat` causing the flicker.


https://github.com/Expensify/App/assets/46707890/48a3bd9e-f0ed-4136-aa61-98dd71a7e6e1

aswin-s commented:
> @aswin-s , For some reason, the background image is flickering, I think first the image is taking `cover` and then changing to `repeat` causing the flicker.
> 
>  Screen.Recording.2023-09-15.at.9.47.48.PM.mov

@abdulrahuman5196 Hmm this seems to be a quirk of react-native-web. It applies the `backgroundSize` style to the image component only after the onLoad event gets triggered. By that time, the image is already rendered on the screen and causes the flicker. Since this is applied at a low level, we can't control the behaviour from a view component.

However we could hide the background image until the image `onLoad` event gets triggerd. This effectively prevents the flicker from happening. Downside is that we need to separate the empty background Image as a separate component and keep the visibility logic in there. It would also keep all the background size calculation logic contained in a  single component rather than duplicated in three places. If this sounds reasonable, I'll apply the fix.

It would look something like this

```jsx

import React from 'react';
import { Image } from 'react-native';
import EmptyStateBackgroundImage from '../../../assets/images/empty-state_background-fade.png';
import CONST from '../../CONST';
import useWindowDimensions from '../../hooks/useWindowDimensions';
import * as StyleUtils from '../../styles/StyleUtils';
import variables from '../../styles/variables';

export default function EmptyStateBackground() {
    const [visible, setVisible] = React.useState(false);
    const {isSmallScreenWidth, windowWidth} = useWindowDimensions();
    const onImageLoaded = () => setVisible(true);
    // If window width is greater than the max background width, repeat the background image
    const maxBackgroundWidth = variables.sideBarWidth + CONST.EMPTY_STATE_BACKGROUND.ASPECT_RATIO * CONST.EMPTY_STATE_BACKGROUND.WIDE_SCREEN.IMAGE_HEIGHT;

    return (
        <Image
            pointerEvents="none"
            onLoadEnd={onImageLoaded}
            source={EmptyStateBackgroundImage}
            style={[StyleUtils.getReportWelcomeBackgroundImageStyle(isSmallScreenWidth), {opacity: visible ? 1 : 0}]}
            resizeMode={windowWidth > maxBackgroundWidth ? 'repeat' : 'cover'}
        />
    );
}


```

And here is the result


https://github.com/Expensify/App/assets/6880914/7af3e9cd-378f-4189-b64a-0d3e94e97f95

aswin-s commented:
@abdulrahuman5196 Any thoughts on above change?

aswin-s committed:
refactor empty state background into a component

aswin-s committed:
Merge branch 'main' into fix/issue-23477

aswin-s committed:
fix: clean lint

aswin-s committed:
Merge branch 'main' into fix/issue-23477

aswin-s commented:
@abdulrahuman5196 As I haven't heard back from you last week, I've gone ahead and made the above change. Please have a look.

abdulrahuman5196 commented:
@aswin-s Sorry I was out couple of days. I am not aligned on the new change. It still looks like a flicker, now its just that from a empty image to new image(contrast to previous image flicker of one size to another size)

>Hmm this seems to be a quirk of react-native-web. It applies the backgroundSize style to the image component only after the onLoad event gets triggered. By that time, the image is already rendered on the screen and causes the flicker. Since this is applied at a low level, we can't control the behaviour from a view component.

How was this information arrived? Is there any GH issue on this on upstream?

aswin-s committed:
fix: add delay in setting opacity

aswin-s committed:
Merge branch 'main' into fix/issue-23477

aswin-s committed:
clean lint

aswin-s commented:
@abdulrahuman5196 
rn-web achieves resizeMode 'repeat' by setting the image as background image for the container element and setting background size on it. To compute the background size it uses the native width and height of the image. The native width and height of the image can not be measured until image is completely loaded. You can refer the background size calculation logic over here.

https://github.com/necolas/react-native-web/blob/a3ea2a0a4fd346a1b32e6cef527878c8e433eef8/packages/react-native-web/src/exports/Image/index.js#L244-L262

Also I've pushed a fix to remove the flicker. I was not able to replicate the flicker on a 4K screen. However flicker appears when developer console was open. So it appears we need to delay the visibility switch until background size calculation completes after image load.

aswin-s committed:
remove comments

abdulrahuman5196 commented:
@aswin-s Even without your latest opacity changes, I am not able to reproduce the flicker anymore. I think it got fixed with the latest merge from main. Can you check again and remove the latest opacity/delay changes?

aswin-s committed:
Merge branch 'main' into fix/issue-23477

aswin-s committed:
fix: revert  changes to prevent flicker

aswin-s commented:
@abdulrahuman5196 Reverted the changes related to opacity and adding delay. I'll still retained `EmptyStateBackground` as a separate component, to avoid repeating the logic at three places.

abdulrahuman5196 commented:
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

<!-- add screenshots or videos here -->




https://github.com/Expensify/App/assets/46707890/9588b8a6-2a3a-4193-81ff-c6d5d5f6fc5c

</details>

<details>
<summary>Mobile Web - Chrome</summary>

<!-- add screenshots or videos here -->



https://github.com/Expensify/App/assets/46707890/efd2d44b-2cb4-4596-b45b-5916da074e4a


</details>

<details>
<summary>Mobile Web - Safari</summary>

<!-- add screenshots or videos here -->




https://github.com/Expensify/App/assets/46707890/199ec9e3-ba74-4a07-937b-883f4b100e1c



</details>

<details>
<summary>Desktop</summary>

<!-- add screenshots or videos here -->





https://github.com/Expensify/App/assets/46707890/a74783ab-97d0-4f1b-a2e8-4dacbefdd948




</details>

<details>
<summary>iOS</summary>

<!-- add screenshots or videos here -->




https://github.com/Expensify/App/assets/46707890/fa9427f2-720d-4c56-9bf9-f1f4ff6c1492

</details>

<details>
<summary>Android</summary>

<!-- add screenshots or videos here -->



https://github.com/Expensify/App/assets/46707890/d9e8c003-ca48-47a1-be2e-aa77685df89b


</details>

abdulrahuman5196 reviewed (approved):
Changes looks good and works well. Reviewers checklist is also complete.

All yours. @grgia 

🎀 👀 🎀
C+ Reviewed

grgia commented on src/components/ReportActionItem/MoneyReportView.js:
```diff
@@ -35,11 +35,7 @@ function MoneyReportView(props) {
-                    source={EmptyStateBackgroundImage}
-                    style={[StyleUtils.getReportWelcomeBackgroundImageStyle(true)]}
-                />
+                <EmptyStateBackground />
```
Can we also use this in the report action thread parent?

aswin-s replied:
@grgia Fixed!

grgia reviewed (changes requested):
Sorry minor change requested- let's also use this in the thread action

aswin-s committed:
fix: use empty state background component

abdulrahuman5196 commented:
@aswin-s This has merge conflicts

aswin-s commented:
Woh! The implementation itself has changed. Now we have an AnimatedBackgroundImage 😓 .

aswin-s committed:
Merge branch 'main' into fix/issue-23477

abdulrahuman5196 commented:
Is this issue fixed then?

aswin-s committed:
fix: use AnimatedEmptyBackground

aswin-s commented:
@abdulrahuman5196 Nope. The issue was still there. Just that there was a new component added named `AnimatedEmptyStateBackground` which uses device sensor data to transform the image slightly when user tilts the device. It internally used the same background image. I removed the new component I added in this PR and merged the fixes into the AnimatedEmptyStateBackground itself.

abdulrahuman5196 commented:
@aswin-s Could you kindly recheck the author's checklist and re-record the screenshots/videos?

aswin-s committed:
fix: prevent unnecessary rerender

aswin-s commented:
@abdulrahuman5196 Re checked the checklist and updated the recordings.

aswin-s commented on src/pages/home/report/AnimatedEmptyStateBackground.js:
```diff
@@ -32,13 +37,14 @@ function AnimatedEmptyStateBackground() {
             transform: [{translateX: withSpring(-IMAGE_OFFSET_X - xOffset.value)}, {translateY: withSpring(yOffset.value)}],
         };
-    });
+    }, []);
```
This prevents unnecessary re-render of background image. Avoiding dependency array causes the style to be recomputed every time component changes. It was causing visual artefacts with resize mode.

abdulrahuman5196 replied:
@aswin-s  Can you attach video of what issue you are facing if this is not present?

Because according to docs it is not required - https://docs.swmansion.com/react-native-reanimated/docs/core/useAnimatedStyle/#arguments

aswin-s replied:
Here you go. Without an empty dependency array the resize mode is not reflected properly.

https://github.com/Expensify/App/assets/6880914/8ebf8aac-7e56-4c6e-9aa2-70d55e013d03

abdulrahuman5196 replied:
Got it. It seems without adding empty dependencies, it is causing issues.

aswin-s commented:
> @aswin-s In author's checklist kindly add `Offline tests` as `Same as tests`

@abdulrahuman5196 Done

abdulrahuman5196 commented:
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

<!-- add screenshots or videos here -->



https://github.com/Expensify/App/assets/46707890/74f7c423-5c90-4c1c-9af8-102fa73d0a92


</details>

<details>
<summary>Mobile Web - Chrome</summary>

<!-- add screenshots or videos here -->



https://github.com/Expensify/App/assets/46707890/67b087ba-e6f4-4d79-955f-bac37e825113



</details>

<details>
<summary>Mobile Web - Safari</summary>

<!-- add screenshots or videos here -->




https://github.com/Expensify/App/assets/46707890/57fa674d-2e4b-4df7-b187-922bea25b478


</details>

<details>
<summary>Desktop</summary>

<!-- add screenshots or videos here -->




https://github.com/Expensify/App/assets/46707890/43d2a6da-2199-4967-9bd2-e9518a3561df


https://github.com/Expensify/App/assets/46707890/f860c405-78be-41b2-8e39-a61a45f890db



</details>

<details>
<summary>iOS</summary>

<!-- add screenshots or videos here -->




https://github.com/Expensify/App/assets/46707890/ee6b073e-845d-43a9-b76a-ea04266a6acc


</details>

<details>
<summary>Android</summary>

<!-- add screenshots or videos here -->




https://github.com/Expensify/App/assets/46707890/08d443ac-aff0-4ac0-9b7d-c7ad0a4ac5e9


</details>

abdulrahuman5196 reviewed (approved):
Changes looks good and works well. Reviewers checklist is also complete.

All yours. @grgia 

🎀 👀 🎀
C+ Reviewed

abdulrahuman5196 commented:
@grgia I have retested the changes since we had to make refractor due to parallel work. Kindly take a look into this.

aswin-s commented:
@grgia Gentle reminder.

grgia reviewed (approved):
Thank you for your patience, I've been working on some internal high priority projects. Great work on this!

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/6480936812) to staging by https://github.com/grgia in version: 1.3.81-4 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/6491054357) to staging by https://github.com/grgia in version: 1.3.83-0 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/6513406442) to production by https://github.com/jasperhuangg in version: 1.3.83-11 🚀

platform | result
---|---
🤖 android 🤖|skipped 🚫
🖥 desktop 🖥|skipped 🚫
🍎 iOS 🍎|skipped 🚫
🕸 web 🕸|skipped 🚫

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/6513406442) to production by https://github.com/jasperhuangg in version: 1.3.83-11 🚀

platform | result
---|---
🤖 android 🤖|skipped 🚫
🖥 desktop 🖥|skipped 🚫
🍎 iOS 🍎|skipped 🚫
🕸 web 🕸|skipped 🚫

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/6513406442) to production by https://github.com/jasperhuangg in version: 1.3.83-11 🚀

platform | result
---|---
🤖 android 🤖|skipped 🚫
🖥 desktop 🖥|skipped 🚫
🍎 iOS 🍎|skipped 🚫
🕸 web 🕸|skipped 🚫

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/6513766880) to production by https://github.com/jasperhuangg in version: 1.3.83-11 🚀

platform | result
---|---
🤖 android 🤖|failure ❌
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|failure ❌
🕸 web 🕸|success ✅
